python/lsst/analysis/tools/actions/plot/matrixPlot.py

```diff
@@ -0,0 +1,348 @@
+        data : `~lsst.analysis.tools.interfaces.KeyedData`
+            The data to plot.
+        **kwargs
+            Additional keyword arguments to pass to the plot.
+
+        Returns
+        -------
+        fig : `~matplotlib.figure.Figure`
+            The resulting figure.
+        """
+        matrix = data[self.matrixKey]
+        if self.vmin is not None and self.vmax is not None:
+            vrange = (self.vmin, self.vmax)
+        else:
+            interval = apViz.PercentileInterval(98.0)
+            vrange = interval.get_limits(np.abs(matrix.flatten()))
+
+        # Allow for the figure object to be passed in.
+        fig = kwargs.get("fig", None)
+        if fig is None:
+            fig = plt.figure(figsize=self.figsize, dpi=300)
+            ax = fig.add_subplot(111)
+        else:
+            ax = fig.gca()
+
+        if self.title:
+            ax.set_title(self.title)
+
+        if self.xAxisLabel:
+            ax.set_xlabel(self.xAxisLabel)
+
+        if self.yAxisLabel:
+            ax.set_ylabel(self.yAxisLabel)
+
+        # Set the colorbar and draw the image.
+        norm = ImageNormalize(vmin=vrange[0], vmax=vrange[1])
+        img = ax.imshow(matrix, interpolation="none", norm=norm)
+
+        # Calculate the aspect ratio of the image.
+        ratio = matrix.shape[0] / matrix.shape[1]
+
+        # Add the colorbar flush with the image axis.
+        fig.colorbar(img, fraction=0.0457 * ratio, pad=0.04, label=self.colorbarLabel)
+
+        if self.centerLabelsBetweenTicks:
+            # Create a dictionary to map dimension labels to their respective
+            # tick values and labels.
+            # Note: We shift by 0.5 to position the tick marks at the pixel
+            # boundaries rather than the center of the pixel.
+            major_tick_data = {
+                "x": (np.array(self.xAxisTickValues) - 0.5, self.xAxisTickLabels),
+                "y": (np.array(self.yAxisTickValues) - 0.5, self.yAxisTickLabels),
+            }
+            for dim, axis in [("x", ax.xaxis), ("y", ax.yaxis)]:
+                # Get the major tick positions and labels.
+                major_tick_positions, major_tick_labels = major_tick_data[dim]
+                # Set major ticks.
+                axis.set_ticks(major_tick_positions, minor=False)
+                # Hide the major tick labels.
+                axis.set_ticklabels("")
+                # Set positions for minor ticks at the midpoints of the major
+                # ticks.
+                axis.set_ticks((major_tick_positions[:-1] + major_tick_positions[1:]) / 2, minor=True)
+                # Set minor tick labels.
+                axis.set_ticklabels(major_tick_labels, minor=True)
+                # Remove minor tick marks for asthetic reasons.
+                axis.set_tick_params(which="minor", length=0)
+        else:
+            # Set the desired tick values and labels if provided.
+            if self.xAxisTickValues is not None:
```

Comment:
Just noting that If it's left like this instead of `if self.xAxisTickValues`, then setting `self.xAxisTickValues=[]` (instead of None) will remove all ticks. I think it's ok to support remove axis ticks and labels but maybe there should be a note about this in the relevant config field docs.

Reply:
Hmm, so you don't want users to be able to turn them off entirely (by supplying an empty list)?

Reply:
Changed it to `if self.xAxisTickValues is not None`; this allows supplying an empty list.